core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -48,49 +45,39 @@ public class NewsCardModel {
     @Default(values = DEFAULT_IMAGE)
     private String image;
 
-    private int likes;
+    private Integer likes = 0;
 
-    private int dislikes;
+    private Integer dislikes = 0;
 
     @Self
     private SlingHttpServletRequest servletRequest;
 
     @SlingObject
     private ResourceResolver resourceResolver;
 
-
     public String getPath() {
         return resource.getPath().replace("/jcr:content", "");
     }
 
-    public String getFullPath() {
-        return resource.getPath();
-    }
-
-
-
     @PostConstruct
     private void init() {
-        String rootPath = resource.getPath();
-        String dislikeParam = servletRequest.getParameter("dislike");
-        String likeParam = servletRequest.getParameter("like");
+        String dislikeParam = servletRequest.getParameter("false");
+        String likeParam = servletRequest.getParameter("true");
         if (likeParam != null) {
-            likes = incrementLikeDislike(likeParam, rootPath);
+            likes = incrementLikeDislike(likeParam);
         } else if (dislikeParam != null) {
-            dislikes = incrementLikeDislike(dislikeParam, rootPath);
+            dislikes = incrementLikeDislike(dislikeParam);
         }
-
+        ValueMap valueMap = resource.adaptTo(ValueMap.class);
+        likes = valueMap.get("like", Integer.class);
+        dislikes = valueMap.get("dislike", Integer.class);
     }
 
-
-    private int incrementLikeDislike(String property, String rootPath) {
-        int incrementProperty = 0;
-        resource = resourceResolver.getResource(rootPath);
-        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
-        if (properties != null) {
-            incrementProperty = properties.get(property, 0);
-            properties.put(property, incrementProperty + 1);
-        }
+    private int incrementLikeDislike(String property) {
+        int incrementProperty;
+        ModifiableValueMap properties = resource.adaptTo(ModifiableValueMap.class);
+        incrementProperty = properties.get(property, 0);
+        properties.put(property, incrementProperty + 1);
```

Comment:
Please notice that you still return a non-incremented value in line 86. This would lead you to an error. But you anyway don't use this value at all (see by previous comment). You assign it in lines 67, 69 and then reassign it in lines 72, 73.
So, you just swept an error under the rug with yet another error))